src/app/[models]/accounts.ts

```diff
@@ -0,0 +1,28 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+export interface IAccount extends Document {
+  provider: string;
+  type: string;
+  providerAccountId: string;
+  access_token: string;
+  expires_at: Date;
+  token_type?: string;
+  id_token?: string;
+  userId: Schema.Types.ObjectId;
+}
+
+const AccountSchema = new mongoose.Schema({
+  userID: { type: String, required: true },
+  provider: { type: String, required: true },
+  type: { type: String, required: true },
+  providerAccountId: { type: String, required: true },
+  access_token: { type: String, required: true },
+  expires_at: { type: Date, required: true },
+  token_type: { type: String, required: false },
+  id_token: { type: String, required: false },
+  userId: { type: Schema.Types.ObjectId, required: true },
+});
+
+const Account = mongoose.model<IAccount>('Account', AccountSchema);
```

Comment:
Lets handle `OverwriteModelError`

```suggestion
const Account = mongoose.models.Account || mongoose.model<IAccount>('Account', AccountSchema);
```